src/puppetlabs/pcp/client.clj

```diff
@@ -325,6 +325,14 @@
     (.start client)
     client))
 
+(defn ^:private append-client-type
+  "Append client type to a ws connection Uri, accounting for possible trailing
+   slash."
+  [url type]
+  (if (= (last url) \/)
```

Comment:
We depend on clojure 1.8, so I think I'll use https://clojure.github.io/clojure/branch-master/clojure.string-api.html#clojure.string/ends-with?.